### PR TITLE
can: overlay and config for reel board CAN expansion

### PR DESCRIPTION
**Pull request by jfischer-no**
Overlay and config for reel board can expansion. Draft for driver and hardware testing only.
@karstenkoenig FYI

### CODE REVIEW

**Review by jukkar — CHANGES_REQUESTED** [review]

**Comment by karstenkoenig** [comment]
This is awesome!

File: samples/drivers/CAN/reel_board.conf [hunk 1]
```diff
@@ -0,0 +1,19 @@
+CONFIG_CAN=y
```
[the remark this answers is not in the document]
Comment (karstenkoenig) [reply samples/drivers/CAN/reel_board.conf]: Sorry for starting with this poor style :-|

**Comment by alexanderwachter** [comment]
@jfischer-phytec-iot can you make a shield for the board?

**Comment by karstenkoenig** [comment]
@alexanderwachter Not sure about the particularities for the reel board as it has a properly defined connector, but I also tried making a general mcp2515 shield before (for the cheap aliexpress breakout boards) and sort of gave up because SPI CS and various other GPIOs doesn't really work well for the shields yet. That's why there is the mcp2515-dts.overlay in the samples so people can at least find a place to start and (fairly) straight forward adjust the cs and irq gpios.

There is a few discussions ongoing such as 
https://github.com/zephyrproject-rtos/zephyr/issues/15637
https://github.com/zephyrproject-rtos/zephyr/pull/14057

but no short term solution. So I think the straightest way ahead is creating a boards folder and tossing the overlays and configs in there until there is a proper solution in place.

I added a board folder here https://github.com/zephyrproject-rtos/zephyr/pull/15658

**Comment by alexanderwachter** [comment]
This PR could be closed. #18081 includes the dts and config.

**Pull request closed without merging**